packages/react-native-bpk-appearance/src/BpkDynamicStyleSheet.js

```diff
@@ -0,0 +1,130 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2016-2019 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+import { StyleSheet } from 'react-native';
+import type {
+  ViewStyle,
+  TextStyle,
+  ImageStyle,
+  TypeForStyleKey,
+} from 'react-native/Libraries/StyleSheet/StyleSheet';
+
+import { type ColorSchemeName } from './BpkAppearance';
+
+type AllStylesProps = {
+  ...$Exact<TextStyle>,
+  ...$Exact<ViewStyle>,
+  ...$Exact<ImageStyle>,
+};
+
+type SenanticColor = { light: string, dark: string };
+type ColorValue<+key: $Keys<AllStylesProps>> =
+  | TypeForStyleKey<key>
+  | SenanticColor;
+
+export type BpkStyleSheetStyle = {|
```

Comment:
This is extending react native's stylesheet type and adding `{ light: string, dark: string }` as a valid option for color props. The create function will parse those props and split them into two `StyleSheets` (light and dark) and this can then be used in code like this:

```
const style = BpkDynamicStyleSheet.create({
  view: {
    shadowColor: { light: 'l', dark: 'd' },
  }
}):
// This will have two props, light and dark
// style.light => { view: { color: 'l' }}
// style.dark => { view: { color: 'd' }}

const MyView = () => {
   // The hook will fetch the correct version based on the current color scheme
   const currentStyle = useBpkDynamicStyleSheet(style);
   return <View style={currentStyle.view} />
}```